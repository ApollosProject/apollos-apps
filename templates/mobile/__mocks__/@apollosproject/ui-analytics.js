export const withTrackOnPress = (Component) => (props) =>
  <Component onPress={() => null} {...props} />;
export const AnalyticsConsumer = ({ children }) =>
  children({ track: () => null, notify: () => null });
