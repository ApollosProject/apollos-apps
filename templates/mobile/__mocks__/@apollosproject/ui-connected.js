import { gql } from '@apollo/client';
import ApollosConfig from '@apollosproject/config';

export const GET_USER_PROFILE = gql`
  query CurrentUserProfile {
    currentUser {
      id
      profile {
        ...UserProfileParts
        campus {
          ...CampusParts
        }
      }
    }
  }
  ${ApollosConfig.FRAGMENTS.CAMPUS_PARTS_FRAGMENT}
  ${ApollosConfig.FRAGMENTS.USER_PROFILE_PARTS_FRAGMENT}
`;

export const UserAvatarUpdate = 'UserAvatarUpdate';
export const usePersonFollowing = () => ({ total: 1 });
export const usePersonFollowedBy = () => ({ total: 1 });
export const useCurrentUserFollowRequests = () => ({ total: 1 });
