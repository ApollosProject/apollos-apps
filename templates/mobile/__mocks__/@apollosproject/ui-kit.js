export const named = () => (Component) => (props) => <Component {...props} />;
export const styled = (style) => (Component) => (props) =>
  <Component {...props} style={style} />;
export const withTheme = (newProps) => (Component) => (props) =>
  <Component {...props} {...newProps} />;
export const withIsLoading = (Component) => (props) =>
  <Component isLoading={false} {...props} />;
export const useTheme = () => ({
  colors: { background: {} },
  sizing: {},
});
export const NavigationService = { resetToAuth: () => null };
export const H1 = 'H1';
export const H2 = 'H2';
export const H3 = 'H3';
export const H4 = 'H4';
export const H5 = 'H5';
export const H6 = 'H6';
export const UIText = 'UIText';
export const Avatar = 'Avatar';
export const AvatarCloud = 'AvatarCloud';
export const Icon = 'Icon';
export const ButtonLink = 'ButtonLink';
export const PaddedView = 'PaddedView';
export const BackgroundView = 'BackgroundView';
export const TableView = 'TableView';
export const Row = 'Row';
export const Divider = 'Divider';
export const Touchable = 'Touchable';
export const TextInput = 'TextInput';
export const Button = 'Button';
export const CampusCard = 'CampusCard';
export const Card = 'Card';
export const CardContent = 'CardContent';
export const ByLine = 'ByLine';
export const AppIcon = 'AppIcon';
export const SocialBar = 'SocialBar';
export const GradientOverlayImage = 'GradientOverlayImage';
export const RadioButton = 'RadioButton';
export const Radio = 'Radio';
export const DateInput = 'DateInput';
