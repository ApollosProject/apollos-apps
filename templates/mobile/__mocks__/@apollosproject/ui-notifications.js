import { createContext } from 'react';

export const requestPermissions = () => null;
export const PushContext = createContext({
  hasPrompted: true,
  hasPushPermission: true,
  loading: true,
  checkPermissions: () => {}, // deprecated, confusing name
  updatePermissionStatus: () => {},
});
