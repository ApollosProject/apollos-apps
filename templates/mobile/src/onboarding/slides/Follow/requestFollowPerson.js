import { gql } from '@apollo/client';

export default gql`
  mutation requestFollow($personId: ID!) {
    requestFollow(followedPersonId: $personId) {
      state
      id
    }
  }
`;
