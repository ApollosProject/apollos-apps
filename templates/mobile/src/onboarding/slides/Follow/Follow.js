import { memo } from 'react';
import PropTypes from 'prop-types';

import { FollowListConnected } from '@apollosproject/ui-connected';
import { named } from '@apollosproject/ui-kit';
import Slide, { SlideContent } from '../../Slide';

const Follow = memo(
  ({
    BackgroundComponent,
    slideTitle,
    description,
    followers,
    primaryNavText,
    ...props
  }) => (
    <Slide primaryNavText={primaryNavText} {...props}>
      {BackgroundComponent}
      <SlideContent title={slideTitle} description={description}>
        <FollowListConnected followers={followers} />
      </SlideContent>
    </Slide>
  )
);

Follow.displayName = 'Follow';

Follow.propTypes = {
  followers: PropTypes.arrayOf(PropTypes.object),
  slideTitle: PropTypes.string,
  description: PropTypes.string,
  primaryNavText: PropTypes.string,
  /* The `Swiper` component used in `<onBoarding>` looks for and hijacks the title prop of it's
   * children. Thus we have to use more unique name.
   */
  /* Recommended usage:
   * - `Image` (react-native)
   * - `GradientOverlayImage` (@apollosproject/ui-kit) for increased readability
   * - `Video` (react-native-video) because moving pictures!
   */
  BackgroundComponent: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

Follow.defaultProps = {
  slideTitle: 'Get connected',
  description: 'Follow others to stay up to date with your church community.',
  followers: [],
  primaryNavText: 'Finish',
};

export default named('ui-onboarding.Follow')(Follow);
