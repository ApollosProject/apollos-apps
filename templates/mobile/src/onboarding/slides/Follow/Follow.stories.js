import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';
import { GradientOverlayImage } from '@apollosproject/ui-kit';

import Follow from './Follow';

storiesOf('ui-onboarding/slides/Follow', module)
  .add('default', () => <Follow />)
  .add('followers', () => (
    <Follow followers={[{ firstName: 'Joe', lastName: 'Schmoe' }]} />
  ))
  .add('slideTitle', () => <Follow slideTitle={'Custom title text'} />)
  .add('description', () => <Follow description={'Custom description text'} />)
  .add('BackgroundComponent', () => (
    <Follow
      BackgroundComponent={
        <GradientOverlayImage
          source={'https://picsum.photos/750/750/?random'}
        />
      }
    />
  ))
  .add('Slide props', () => <Follow onPressPrimary={() => {}} />);
