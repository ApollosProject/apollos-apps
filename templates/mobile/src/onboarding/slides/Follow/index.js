import Follow from './Follow';

export FollowConnected from './FollowConnected';

export default Follow;
