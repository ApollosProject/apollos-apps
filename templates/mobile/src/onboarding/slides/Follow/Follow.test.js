import renderer from 'react-test-renderer';
import { GradientOverlayImage } from '@apollosproject/ui-kit';

import Follow from './Follow';

describe('The Onboarding Follow component', () => {
  it('should render', () => {
    const tree = renderer.create(<Follow />);
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom first name', () => {
    const tree = renderer.create(<Follow firstName={'firstName'} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom title', () => {
    const tree = renderer.create(<Follow slideTitle={'Custom title text'} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom description', () => {
    const tree = renderer.create(
      <Follow description={'Custom description text'} />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a BackgroundComponent', () => {
    const tree = renderer.create(
      <Follow
        BackgroundComponent={
          <GradientOverlayImage
            source={'https://picsum.photos/375/812/?random'}
          />
        }
      />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should pass additional props to Slide', () => {
    const tree = renderer.create(<Follow onPressPrimary={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});
