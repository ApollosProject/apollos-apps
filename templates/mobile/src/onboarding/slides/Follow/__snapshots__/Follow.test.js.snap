// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Onboarding Follow component should pass additional props to Slide 1`] = `
Array [
  <RCTScrollView
    contentContainerStyle={
      Object {
        "minHeight": "100%",
      }
    }
    overScrollMode="auto"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <SafeAreaView
        style={[Function]}
      >
        <View>
          <View
            style={[Function]}
          >
            <H2
              style={[Function]}
            >
              Get connected
            </H2>
          </View>
          <H5
            style={[Function]}
          >
            Follow others to stay up to date with your church community.
          </H5>
        </View>
        <FollowListConnected
          followers={Array []}
        />
      </SafeAreaView>
    </View>
  </RCTScrollView>,
  <PaddedView
    style={[Function]}
    vertical={false}
  >
    <SafeAreaView
      edges={
        Array [
          "bottom",
        ]
      }
    >
      <Button
        onPress={[MockFunction]}
        style={Object {}}
      >
        <H5>
          Finish
        </H5>
        <Icon
          name="arrow-next"
        />
      </Button>
    </SafeAreaView>
  </PaddedView>,
]
`;

exports[`The Onboarding Follow component should render 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Get connected
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          Follow others to stay up to date with your church community.
        </H5>
      </View>
      <FollowListConnected
        followers={Array []}
      />
    </SafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`The Onboarding Follow component should render a BackgroundComponent 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <GradientOverlayImage
      source="https://picsum.photos/375/812/?random"
    />
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Get connected
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          Follow others to stay up to date with your church community.
        </H5>
      </View>
      <FollowListConnected
        followers={Array []}
      />
    </SafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`The Onboarding Follow component should render a custom description 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Get connected
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          Custom description text
        </H5>
      </View>
      <FollowListConnected
        followers={Array []}
      />
    </SafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`The Onboarding Follow component should render a custom first name 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  firstName="firstName"
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Get connected
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          Follow others to stay up to date with your church community.
        </H5>
      </View>
      <FollowListConnected
        followers={Array []}
      />
    </SafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`The Onboarding Follow component should render a custom title 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Custom title text
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          Follow others to stay up to date with your church community.
        </H5>
      </View>
      <FollowListConnected
        followers={Array []}
      />
    </SafeAreaView>
  </View>
</RCTScrollView>
`;
